libia2/include/ia2.h

```diff
@@ -141,7 +141,21 @@ asm(".macro mov_pkru_eax pkey\n"
         .pkey = n,                                                             \
         .address = &init_pkey_##n##_ctor,                                      \
     };                                                                         \
+    __asm__ volatile(/* Set PKRU to the compartment's value */                 \
+                     "xorl %%ecx, %%ecx\n"                                     \
+                     "xorl %%edx, %%edx\n"                                     \
+                     "mov_pkru_eax " #n "\n" IA2_WRPKRU "\n"                   \
+                     :                                                         \
+                     :                                                         \
+                     : "rax", "rcx", "rdx");                                   \
     dl_iterate_phdr(protect_pages, &args);                                     \
+    __asm__ volatile(/* Set PKRU to the default value */                       \
+                     "xorl %%ecx, %%ecx\n"                                     \
+                     "xorl %%edx, %%edx\n"                                     \
+                     "xorl %%eax, %%eax\n" IA2_WRPKRU "\n"                     \
```

Comment:
I'm not sure "default value" is the right way to describe this--we're zeroing PKRU, which isn't the system's default.

I would say "Set PKRU to fully untrusted".